### PR TITLE
Feature/templates

### PR DESCRIPTION
**Pull request by peerfikhman**
Hey 
Made some changes to front page of templates please check it out. mostly done on separate files: templatesScript and templatesForm.

### CODE REVIEW

**Review by shir149 — APPROVED** [review]
no comment. except changes

File: Front/js/templatesScript.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        if (delTempConfirm) {
+            fetch('http://localhost:3000/templates',
+                {
+                    method: 'DELETE',
+                    mode: `cors`,
+                    headers: {'Accept': `application/json`},
+                    body: JSON.stringify({"template_id": template.template_id})
+                }).then((response) => {
+                    alert(`Template ${template.name} deleted successfully`)
+                    window.location.reload();
```
Comment (DaveAvigad) [inline Front/js/templatesScript.js:47]: `window.location.reload();`
why?
seems like a workaorund

File: Front/style.css [hunk 2]
```diff
@@ -220,6 +220,7 @@ body{
     border-bottom-width: 4px !important;
```
Comment (DaveAvigad) [inline Front/style.css:220]: using `!important` is a bad practice, avoid this.

File: README.md [hunk 3]
```diff
@@ -1 +1,34 @@
-# Mail-Service
\ No newline at end of file
+# Mail-Service
```
Comment (DaveAvigad) [inline README.md:1]: 🎉

File: README.md [hunk 4]
```diff
@@ -1 +1,34 @@
...
+
+Welcome to our Mail- Service !
+Here we manage email templates, and send out your custom emails for now or for later. You can write your own template by editing the HTML code, save, create and delete the templates. You can choose to send a new E-mail, with all the usual information (CC, BCC, subject, to). As for the body you can either type in plain text, or choose a template, and then also choose to edit the template HTML before you send to customize further. You may choose to send out the email right away, or schedule it for a future time. Later you can see stats about the emails sent, the future emails, and the templates.
+Before you start:
```
Comment (DaveAvigad) [inline README.md:5]: this file should be according to the md file best practices, please [read this](https://www.freecodecamp.org/news/how-to-write-a-good-readme-file/) and follow [this template](https://github.com/nuxeo/nuxeo/blob/master/README.md).

File: Routers/clientRouter.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+
+const clientController = require('../controllers/clientController');
```
Comment (DaveAvigad) [inline Routers/clientRouter.js:3]: `clientController`
use meaningful names

File: Routers/templatesRouter.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+const express = require('express');
```
Comment (DaveAvigad) [inline Routers/templatesRouter.js:1]: ✅ readability
✅ clear & clean
✅ modularity

File: controllers/clientController.js [hunk 7]
```diff
@@ -10,7 +10,8 @@ function errorHandler(req, res) {
 function loadPage(req, res) {
     const pathName = new URL(req.url, `http://${req.headers.host}`).pathname;
     res.writeHead(200);
-    fs.createReadStream(process.cwd() + '/Front' + pathName)
+    // res.status(200);
```
Comment (DaveAvigad) [inline controllers/clientController.js:13]: remove comments before deployment

File: controllers/templatesControllers.js [hunk 8]
```diff
@@ -1,34 +1,51 @@
...
+const {getNumOfSentEmails} = require("../DAL/emailsDAL");
 
-
+// ************* HTTP way of handling id passed in query string ******************/
 getTemplateID = (req) => {
     return new URL(req.url, `http://${req.headers.host}`)
         .searchParams.get('id');
 }
 
+// ************* Express way of handling Id passed in path params  ****************/
```
Comment (DaveAvigad) [inline controllers/templatesControllers.js:12]: do not push commented-out code

File: controllers/templatesControllers.js [hunk 9]
```diff
@@ -1,34 +1,51 @@
...
+// getTemplateID = (req) => {
+//     return  req.params.id;
+// }
+
+
+
 function getAllTemplates(req, res) {
     res.setHeader('Access-Control-Allow-Origin', '*');
```
Comment (DaveAvigad) [inline controllers/templatesControllers.js:20]: DRY
consider adding some middleware or handler that enrich the response with the desired headers

File: controllers/templatesControllers.js [hunk 10]
```diff
@@ -39,7 +56,7 @@ function editTemplateHandler(req, res) {
     req
         .on(`data`, data => editedTemplate = JSON.parse(data.toString()))
```
Comment (DaveAvigad) [inline controllers/templatesControllers.js:57]: your mixing layers and technologies here
```
req.on('data', ...).on('end' ...)..
```
is a node HTTP approach, while you are using Express here, please read how to get the body in  Express.

File: server.js [hunk 11]
```diff
@@ -1,6 +1,10 @@
 let http = require('http');
 const routes = require(`./router`);
 require('dotenv').config();
-
+// ******** Http Server **************//
 http.createServer(routes).listen(process.env.PORT);
```
Comment (DaveAvigad) [inline server.js:5]: please switch to Express server in this assignment

File: controllers/templatesControllers.js (hunk 10 above)
Comment (DaveAvigad) [reply controllers/templatesControllers.js:57]: sorry my bad, this comment is not relevant to you as these changes do not occur by this commit.

File: server.js [hunk 12]
```diff
@@ -1,6 +1,10 @@
...
 console.log("server running on port 3000");
+
+//************* Express Server ************//
+// const {app} = require('./Routers/mainRouter');
+// app.listen(process.env.PORT);
```
Comment (DaveAvigad) [inline server.js:10]: remove comments

File: services/templates.js [hunk 13]
```diff
@@ -32,7 +36,7 @@ function create(data) {
 
 }
 
-function edit(templateID, data) {
+function editTemplate(templateID, data) {
     const {name = "", creator = "", html = ""} = data;
```
Comment (DaveAvigad) [inline services/templates.js:40]: please use Linter in your project. [read more about ESlint](https://eslint.org/)

**Review by DaveAvigad — COMMENTED** [review]
looks good,
- the variable name should be meaningful 
- DRY do not repeat yourself - too much code duplication
- you are using Express and also HTTP, do not mix these two approaches
- the redirection should be handled by the server
- please add linter
- do not use inline css

great work! 🚀

**Pull request merged**